compiler/src/main/kotlin/io/github/aplcornell/viaduct/backends/cleartext/CleartextCircuitCodeGenerator.kt

```diff
@@ -112,6 +121,128 @@ class CleartextCircuitCodeGenerator(context: CodeGeneratorContext) : AbstractCod
         }
     }
 
+    private fun createCommitment(
+        source: Protocol,
+        target: Protocol,
+        argument: Argument,
+        builder: CodeBlock.Builder,
+    ): CodeBlock {
+        require(context.host in source.hosts + target.hosts)
+        if (source !is Local) {
+            throw UnsupportedCommunicationException(source, target, argument.sourceLocation)
+        }
+        require(source.hosts.size == 1 && source.host in source.hosts)
+        require(target is CommitmentProtocol)
+        if (target.cleartextHost != source.host || target.cleartextHost in target.hashHosts) {
+            throw UnsupportedCommunicationException(source, target, argument.sourceLocation)
+        }
+
+        val argType = kotlinType(argument.type.shape, typeTranslator(argument.type.elementType.value))
+        val sendingHost = target.cleartextHost
+        val receivingHosts = target.hashHosts
+        return when (context.host) {
+            sendingHost -> {
+                val tempName1 = context.newTemporary("CommitTemp")
```

Comment:
better to name these "committed" and "commitment". Giving them the same name isn't great. Also, Kotlin variables names should be lowercase.